src/main/java/org/openstreetmap/atlas/checks/validation/tag/MixedCaseNameCheck.java

```diff
@@ -0,0 +1,263 @@
+        this.checkNameCountries = (List<String>) configurationValue(configuration,
+                "check_name.countries", CHECK_NAME_COUNTRIES_DEFAULT);
+        this.languageNameTags = (List<String>) configurationValue(configuration,
+                "name.language.keys", LANGUAGE_NAME_TAGS_DEFAULT);
+        this.lowerCasePrepositions = (List<String>) configurationValue(configuration,
+                "name.prepositions", LOWER_CASE_PREPOSITIONS_DEFAULT);
+        this.lowerCaseArticles = (List<String>) configurationValue(configuration, "name.articles",
+                LOWER_CASE_ARTICLES_DEFAULT);
+        this.splitCharacters = (String) configurationValue(configuration, "regex.split",
+                SPLIT_CHARACTERS_DEFAULT);
+        this.nameAffixes = (String) configurationValue(configuration, "name.affixes",
+                NAME_AFFIXES_DEFAULT, value -> String.join("|", (List<String>) value));
+        this.mixedCaseUnits = (String) configurationValue(configuration, "name.units",
+                MIXED_CASE_UNITS_DEFAULT, value -> String.join("|", (List<String>) value));
+    }
+
+    /**
+     * This function will validate if the supplied atlas object is valid for the check.
+     *
+     * @param object
+     *            the atlas object supplied by the Atlas-Checks framework for evaluation
+     * @return {@code true} if this object should be checked
+     */
+    @Override
+    public boolean validCheckForObject(final AtlasObject object)
+    {
+        // Valid objects are items that were OSM nodes or ways
+        return !(object instanceof Relation) && !this.isFlagged(object.getOsmIdentifier())
+                && ((object.getTags().containsKey(ISOCountryTag.KEY)
+                        // Must have an ISO code that is in checkNameCountries...
+                        && this.checkNameCountries
+                                .contains(object.tag(ISOCountryTag.KEY).toUpperCase())
+                        // And have a name tag
+                        && Validators.hasValuesFor(object, NameTag.class))
+                        // Or it must have a specific language name tag from languageNameTags
+                        || this.languageNameTags.stream()
+                                .anyMatch(key -> object.getOsmTags().containsKey(key)));
+    }
+
+    /**
+     * This is the actual function that will check to see whether the object needs to be flagged.
+     *
+     * @param object
+     *            the atlas object supplied by the Atlas-Checks framework for evaluation
+     * @return an optional {@link CheckFlag} object that
+     */
+    @Override
+    protected Optional<CheckFlag> flag(final AtlasObject object)
+    {
+        final List<String> mixedCaseNameTags = new ArrayList<>();
+        final Map<String, String> osmTags = object.getOsmTags();
+
+        // Check ISO against list of countries for testing name tag
+        if (this.checkNameCountries.contains(object.tag(ISOCountryTag.KEY).toUpperCase())
```

Comment:
Again same comments as above. Not necessary, as it is all handled by the BaseCheck.